app/models/account.rb

```diff
@@ -594,5 +594,8 @@ def master_uniqueness
     errors.add :master, 'can be only one' if scope.exists?(master: true)
   end
 
-  protected
+  reset_tenant_id do
```

Comment:
Maybe we could find a name for that method that makes it clear that reset tenant_id will be performed when this is loaded, like `config_ reset_tenant_id` or `define_reset_tenant_id`. WDYT?